### PR TITLE
 Backward-compat support setting proxy via environment variables

### PR DESCRIPTION
**Pull request by kutsen99**
The [doc](https://github.com/MicrosoftDocs/azure-devops-docs/blob/master/docs/pipelines/agents/proxy.md) says the proxy can be set by the env variables, but it doesn't work for the agent plugin.
Add this backward-compat support so that it won't break the customer who uses this way of proxy configuration in pipeline artifact task.

VSTS_HTTP_PROXY -> proxy url
VSTS_HTTP_PROXY_USERNAME -> username
VSTS_HTTP_PROXY_PASSWORD -> password

### CODE REVIEW

File: src/Agent.Sdk/TaskPlugin.cs [hunk 1]
```diff
@@ -325,6 +325,21 @@ public AgentWebProxySettings GetProxyConfiguration()
...
+                var ProxyUrl = Environment.GetEnvironmentVariable("VSTS_HTTP_PROXY");
+                ProxyUrl = ProxyUrl.Trim();
+                var ProxyUsername = Environment.GetEnvironmentVariable("VSTS_HTTP_PROXY_USERNAME");
+                var ProxyPassword = Environment.GetEnvironmentVariable("VSTS_HTTP_PROXY_PASSWORD");
+                return new AgentWebProxySettings()
+                {
+                    ProxyAddress = ProxyUrl,
+                    ProxyUsername = ProxyUsername,
+                    ProxyPassword = ProxyPassword,
+                    WebProxy = new AgentWebProxy(proxyUrl, ProxyUsername, ProxyPassword, null)
```
Comment (fadnavistanmay) [inline src/Agent.Sdk/TaskPlugin.cs:340]: We don't need proxy bypass list?

**Review by fadnavistanmay — APPROVED** [review]

**Comment by fadnavistanmay** [comment]
@mjroghelia - can you please review. Thanks

File: src/Agent.Sdk/TaskPlugin.cs (hunk 1 above)
Comment (kutsen99) [reply src/Agent.Sdk/TaskPlugin.cs:340]: The old fashioned way doesn't have a corresponding variable for that, do I assume the customer won't need that.

**Review by mjroghelia — APPROVED** [review]

**Pull request merged**